lex/lexer.mll

```diff
@@ -293,7 +293,7 @@ and comment = parse
       string lexbuf;
       reset_string_buffer();
       comment lexbuf }
-  | '{' (['a'-'z' '_'] * as delim) '|'
+  | '{' ('%' '%'? identstart identbody* ('.' identstart identbody*)* [' ' '\009' '\012']*)? (['a'-'z' '_']* as delim) "|"
```

Comment:
Could you use/copy the regexes for identifier and whitespaces from the OCaml lexer instead ? It should use the exact same one, so that it's easier to check.